### DISCUSSION TITLE
Some objects appears multiple time in my room scan, using Rtambap

### DISCUSSION THREAD

**Discussion by damavand1** [introlab/rtabmap#1195] [General]
Hi

This is my room scan, and I'm showing my room from top view in this picture
as you see, Rtabmap scanned 3 times my brown desk in different positions,
- is it an issue ?
- is it possible this made some problems in localization mode ?
- is it better rtabmab merge all by merging this together to make clean scan ?  

![Screenshot 2024-01-04 114931](https://github.com/introlab/rtabmap/assets/104978921/40169f04-8922-455d-af06-af6509a86180)

tnx

**Comment by matlabbe** [comment]
Which sensor did you use?

This drift would be corrected only if loop closures can be detected.

**Reply by damavand1** [reply]
I used kinect xbox 360
I have to mention that this scan has been obtained using RTAB-Map on the Windows OS.

the important question is why loop closures don't works ?

**Reply by matlabbe** [reply]
Loop closure detection requires discriminative visual features to work correctly. Textureless surfaces won't trigger any loop closures. Note also that if the camera is looking again at the same thing but with a very different point of view, the default features may not work properly. In that case, we should use features more invariant to rotation/scale/shear like SIFT (e.g., with parameter `Kp/DetectorStrategy=1` if rtabmap is built with latest opencv version or with opencv_contrib's nonfree module for older opencv versions).
